File: backend/Data/Entities/BaseEntity.cs
using System;

namespace Pims.Api.Data.Entities
{
    /// <summary>
    /// BaseEntity class, provides a way to inherit shared properties on all entities.
    /// </summary>
    public abstract class BaseEntity
    {
        #region Properties

        /// <summary>
        /// get/set - The foreign key to the user who created this entity.
        /// </summary>
        /// <value></value>
        public Guid? CreatedById { get; set; }

        /// <summary>
        /// get/set - When this entity was created.
        /// </summary>
        /// <value></value>
        public DateTime CreatedOn { get; set; }

        /// <summary>
        /// get/set - Who updated this entity last.
        /// </summary>
        /// <value></value>
        public Guid? UpdatedById { get; set; }

        /// <summary>
        /// get/set - When this entity was updated.
        /// </summary>
        /// <value></value>
        public DateTime? UpdatedOn { get; set; }

        /// <summary>
        /// get/set - The concurrency row version.
        /// </summary>
        /// <value></value>
        public byte[] RowVersion { get; set; }
        #endregion

        #region Constructors
        /// <summary>
        /// Creates a new instance of a BaseEntity class.
        /// Initializes the default values.
        /// /// </summary>
        public BaseEntity ()
        {
            this.CreatedOn = DateTime.UtcNow;
        }
        #endregion
    }
}

File: express-api/src/typeorm/Entities/BaseEntity.ts
import { Users } from '@/typeorm/Entities/Users';
import { Column, CreateDateColumn, ManyToOne, JoinColumn, Index } from 'typeorm';

export abstract class BaseEntity {
  @ManyToOne(() => Users, (user) => user.Id)
  @JoinColumn({ name: 'CreatedById' })
  @Index()
  CreatedById: Users;

  @CreateDateColumn()
  CreatedOn: Date;

  @ManyToOne(() => Users, (user) => user.Id, { nullable: true })
  @JoinColumn({ name: 'UpdatedById' })
  @Index()
  UpdatedById: Users;

  @Column({ type: 'timestamp', nullable: true })
  UpdatedOn: Date;
}
